### ISSUE TITLE
ansible-vault fails to decrypt to stdout

### ISSUE THREAD

**Issue by cornelius-keller** [ansible/ansible#32168]
##### ISSUE TYPE

 - Bug Report
##### COMPONENT NAME
 ansible-vault 
##### ANSIBLE VERSION

```
ansible 2.4.0.0
  config file = /etc/ansible/ansible.cfg
  configured module search path = [u'/root/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python2.7/dist-packages/ansible
  executable location = /usr/bin/ansible
  python version = 2.7.12 (default, Nov 19 2016, 06:48:10) [GCC 5.4.0 20160609]

```

##### CONFIGURATION
DEFAULT_VAULT_PASSWORD_FILE(env: ANSIBLE_VAULT_PASSWORD_FILE) = /test

##### OS / ENVIRONMENT

running in ubuntu 16.04
not managing any os.

##### SUMMARY

From the output of `ansible-vault --help decrypt
```
  --output=OUTPUT_FILE  output file name for encrypt or decrypt; use - for
                        stdout
```
When using `ansible-vault decrypt` with `--output=-` or `--output -` it decrypts to a file named `-` instead to stdout as expected. 

##### STEPS TO REPRODUCE

Create  Dockerfile
```Dockerfile
FROM ubuntu:16.04

RUN apt-get update
RUN apt-get install -y software-properties-common
RUN apt-add-repository ppa:ansible/ansible
RUN apt-get update
RUN apt-get install -y ansible

ENV ANSIBLE_VAULT_PASSWORD_FILE=test
RUN echo "test" > test
RUN echo "test" > test.txt
RUN ansible-vault encrypt test.txt
RUN ansible-vault decrypt test.txt --output -
RUN cat ./-
```

run:
```sh
docker build -t ansible-vault-test .
docker run -it --rm ansible-vault-test ansible-vault decrypt --output - test.txt
docker run -it --rm ansible-vault-test  cat -- ./-
```

##### EXPECTED RESULTS

Breefly:
ansible-vault writes the decrypted file to stdout.

TLDR:
run:
```sh
docker build -t ansible-vault-test .
docker run -it --rm ansible-vault-test ansible-vault decrypt --output - test.txt
```
expected result (f.e using ansible 2.3.x ):
```
test
Decryption successful
```

actual result:
```
Decryption successful
```
run:
```sh
docker run -it --rm ansible-vault-test  cat -- ./-
```
expected result:
```
cat: ./-: No such file or directory
```
actual result
```
test
```

##### ACTUAL RESULTS

ansible-vault writes the decrypted file to a file named `-`

```

```

**Comment by cornelius-keller** [comment]
Sorry I did not see but this is a duplicate of https://github.com/ansible/ansible/issues/30550
